src/schema/resolvers.js

```diff
@@ -250,124 +273,245 @@ module.exports = {
+      if (await isAllowanceEnough(senderAddress, topicAddress, amount)) {
+        approveAmount = amount;
+        type = 'APPROVESETRESULT';
+      } else {
+        approveAmount = 0;
+        type = 'RESETAPPROVESETRESULT';
+      }
+
+      // Send approve tx
+      let txid;
       try {
-        const resp = await fetch('http://localhost:5555/approve', {
-          method: 'POST',
-          body: JSON.stringify({
-            spender: oracleAddress,
-            value: consensusThreshold,
-            senderAddress,
-          }),
-          headers: { 'Content-Type': 'application/json' },
-        }).then(res => res.json());
-
-        approveTxid = resp.result.txid;
+        const tx = await bodhiToken.approve({
+          spender: topicAddress,
+          value: approveAmount,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error call /approve: ${err.message}`);
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
       const tx = {
-        _id: approveTxid,
+        _id: txid,
+        txid,
         version,
-        type: 'APPROVESETRESULT',
-        txStatus: 'PENDING',
+        type,
+        status: 'PENDING',
         senderAddress,
-        entityId: oracleAddress,
-        optionIdx: resultIdx,
-        token: 'QTUM',
-        amount: consensusThreshold,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'BOT',
+        amount,
         createdTime: Date.now().toString(),
       };
+      await DBHelper.insertTransaction(Transactions, tx);
+
+      return tx;
+    },
+
+    createVote: async (root, data, { db: { Transactions } }) => {
+      const {
+        version,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        amount,
+        senderAddress,
+      } = data;
+
+      // Make sure allowance is 0, or it needs to be reset
+      let approveAmount;
+      let type;
+      if (await isAllowanceEnough(senderAddress, topicAddress, amount)) {
+        approveAmount = amount;
+        type = 'APPROVEVOTE';
+      } else {
+        approveAmount = 0;
+        type = 'RESETAPPROVEVOTE';
+      }
 
+      // Send approve tx
+      let txid;
       try {
-        await Transactions.insert(tx);
+        const tx = await bodhiToken.approve({
+          spender: topicAddress,
+          value: approveAmount,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error insert Transactions: ${err.message}`);
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
+      const tx = {
+        _id: txid,
+        txid,
+        version,
+        type,
+        status: 'PENDING',
+        senderAddress,
+        topicAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'BOT',
+        amount,
```

Comment:
convert amount from `decimalstring` to `hexstring`

Reply:
will leave as decimal string as discussed.

Reply:
as discussed, converted from `hexString` to `decimalString` in model layer. decimalString already being passed in as mutation.